zipkin-server/src/main/java/zipkin/server/ActuateStorageMetrics.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.server;
+
+import org.springframework.boot.actuate.endpoint.PublicMetrics;
+import org.springframework.boot.actuate.metrics.Metric;
+import org.springframework.boot.actuate.metrics.buffer.BufferMetricReader;
+import org.springframework.boot.actuate.metrics.buffer.CounterBuffers;
+import org.springframework.boot.actuate.metrics.buffer.GaugeBuffers;
+import zipkin.storage.StorageMetrics;
+import zipkin.internal.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static zipkin.internal.Util.checkNotNull;
+
+/**
+ * The following are exported to the "/metrics" endpoint:
+ *
+ * <pre>
+ * <ul>
+ *     <li>counter.storage.traces.retained.$type - Number of traces available for query</li>
+ *     <li>counter.storage.spans.retained.$type - Number of spans available for query.  Each span
+ *     will belong to a trace</li>
+ *     <li>counter.storage.traces.purged.$type - Number of traces that have been deleted</li>
+ *     <li>counter.storage.spans.purged.$type - Number of spans that have been deleted</li>
+ * </ul>
+ * </pre>
+ *
+ * See https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-metrics.html
+ */
+public final class ActuateStorageMetrics implements StorageMetrics, PublicMetrics
```

Comment:
I like the idea of what you are doing, but I don't want us to add a top-level type that only supports in-memory storage. It adds a significant cognitive and maintenance impact in support of a non-production configuration.

If this is needed, I would probably flatten the hierarchy and expose as an explicit hook for InMemoryStorage (without exposing a new public type). If not needed beyond tests, I would remove it entirely. 

Keep in mind we don't even have counters like this for production code such as cassandra cache counters!